packages/react/src/components/AnchoredOverlay/index.tsx

```diff
@@ -128,11 +133,24 @@ const AnchoredOverlay = forwardRef(
       }
     }, [onShiftChange, middlewareData.shift?.x, middlewareData.shift?.y]);
 
-    return (
+    const AnchoredOverlayComponent = (
       <Component ref={ref} {...props} style={{ ...floatingStyles, ...style }}>
         {children}
       </Component>
     );
+
+    if (portal && !isBrowser()) {
+      return null;
+    }
+
+    return portal && typeof portal !== 'undefined'
+      ? (createPortal(
+          AnchoredOverlayComponent,
+          (portal && 'current' in portal ? portal.current : portal) ||
+            // eslint-disable-next-line ssr-friendly/no-dom-globals-in-react-fc
+            document?.body
+        ) as React.ReactPortal)
+      : AnchoredOverlayComponent;
```

Comment:
It is easier to read this code if we set a const for the portal element like we do for the dialog component but that may be my opinion.
```
if (portal && typeof portal !== 'undefined') {
      const portalElement = 'current' in portal ? portal.current : portal;
      return createPortal(
        AnchoredOverlayComponent,
         // eslint-disable-next-line ssr-friendly/no-dom-globals-in-react-fc
        portalElement || document?.body
      ) as React.ReactPortal;
    }
    
    return AnchoredOverlayComponent;
```

Reply:
This does work slightly differently than dialog since the portal is optional, but yeah we could return early here. Let me make that change.

Reply:
Good callout, this actually allowed for us to remove the cast `as React.Portal`.

Reply:
I meant that making a `const` like `const portalElement = 'current' in portal ? portal.current : portal;` would be easier to read.